app/controllers/api/v1/accounts/articles_controller.rb

```diff
@@ -0,0 +1,38 @@
+class Api::V1::Accounts::ArticlesController < Api::V1::Accounts::BaseController
+  before_action :portal
+  before_action :fetch_article, except: [:index, :create]
+
+  def index
+    @articles = @portal.articles.where(status: :published)
+    @articles.search(params[:payload]) if params[:payload].present?
+  end
+
```

Comment:
Please add :show action as well.

Reply:
I did it because we only did the show with edit action for automation pages.

Reply:
We need an individual article route as well. We are not showing the information in a modal, instead, it is shown on a different page, fetching all the articles is not ideal in that case.